test/extensions/filters/http/lua/lua_integration_test.cc

```diff
@@ -1331,5 +1335,125 @@ name: lua
   EXPECT_TRUE(response->complete());
 }
 
+// Forward declare the filter class
+class TestTypedMetadataFilter;
+
+// Custom filter to add typed metadata
+class TestTypedMetadataFilter : public Network::ReadFilter {
+public:
+  Network::FilterStatus onData(Buffer::Instance&, bool) override {
+    return Network::FilterStatus::Continue;
+  }
+
+  Network::FilterStatus onNewConnection() override {
+    const std::string metadata_key = "envoy.test.typed_metadata";
+
+    // Get mutable access to the typed filter metadata map
+    auto& typed_filter_metadata = *read_callbacks_->connection()
+                                       .streamInfo()
+                                       .dynamicMetadata()
+                                       .mutable_typed_filter_metadata();
+
+    // Create metadata protobuf
+    envoy::data::core::v3::TlvsMetadata metadata;
+    metadata.mutable_typed_metadata()->insert({"test_key", "test_value"});
+
+    // Pack metadata into Any
+    ProtobufWkt::Any typed_metadata;
+    typed_metadata.PackFrom(metadata);
+
+    // Insert into typed filter metadata map
+    typed_filter_metadata[metadata_key] = typed_metadata;
+
+    return Network::FilterStatus::Continue;
+  }
+
+  void initializeReadFilterCallbacks(Network::ReadFilterCallbacks& callbacks) override {
+    read_callbacks_ = &callbacks;
+  }
+
+private:
+  Network::ReadFilterCallbacks* read_callbacks_{};
+};
+
+// Filter factory
+class TestTypedMetadataFilterConfig
+    : public Server::Configuration::NamedNetworkFilterConfigFactory {
+public:
+  absl::StatusOr<Network::FilterFactoryCb>
+  createFilterFactoryFromProto(const Protobuf::Message&,
+                               Server::Configuration::FactoryContext&) override {
+    return Network::FilterFactoryCb([](Network::FilterManager& filter_manager) -> void {
+      filter_manager.addReadFilter(std::make_shared<TestTypedMetadataFilter>());
+    });
+  }
+
+  ProtobufTypes::MessagePtr createEmptyConfigProto() override {
+    return std::make_unique<ProtobufWkt::Any>();
+  }
+
+  std::string name() const override { return "envoy.test.typed_metadata"; }
+  std::set<std::string> configTypes() override { return {}; };
+};
+
+TEST_P(LuaIntegrationTest, ConnectionTypedMetadata) {
+  // Register our test filter
+  TestTypedMetadataFilterConfig factory;
+  Registry::InjectFactory<Server::Configuration::NamedNetworkFilterConfigFactory> registry(factory);
+
+  // Setup network filter config
+  const std::string FILTER_CONFIG = R"EOF(
+name: envoy.test.typed_metadata
+typed_config:
+  "@type": type.googleapis.com/google.protobuf.Any
```

Comment:
I think the integration test should use some example data from the proxy protocol TLVs, since that's your goal with this change, and make sure the lua can get all the fields you'd need.